kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/partition/TestPartitionerProjection.java

```diff
@@ -270,6 +270,48 @@ public void run() {
   }
 
   @Test
+  public void testLongFixedSizeRangeFieldPartitionerRangePredicate() {
+    final FieldPartitioner<Long, Long> fp =
+        new LongFixedSizeRangeFieldPartitioner("num", 5);
+    Assert.assertEquals(Ranges.closed(-5L, 10L),
+        fp.project(Ranges.open(-2L, 15L)));
+    Assert.assertEquals(Ranges.closed(5L, 10L),
+        fp.project(Ranges.open(5L, 15L)));
+    Assert.assertEquals(Ranges.closed(5L, 10L),
+        fp.project(Ranges.open(4L, 15L)));
+    Assert.assertEquals(Ranges.closed(0L, 15L),
+        fp.project(Ranges.closed(4L, 15L)));
+    Assert.assertEquals(Ranges.closed(5L, 20L),
+        fp.project(Ranges.openClosed(5L, 21L)));
```

Comment:
It would be good to have some ranges with just one bound tested with project, too.